src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,47 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_STORAGE_SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_STORAGE_SIZE];
+        values = (V[]) new Object[MAX_STORAGE_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int i;
+        for (i = 0; i < size; i++) {
```

Comment:
```suggestion
        for (int i = 0; i < size; i++) {
```
You only need the i variable in the loop, outside the loop use size